### ISSUE TITLE
Investigate alternatives to `@microsoft/api-extractor`

### ISSUE THREAD

**Issue by ifiokjr** [remirror/remirror#481]
### Description

The extractor is too slow when run in a large monorepo. It takes more than 2 minutes to extract and check if the API has changed across the whole project. It also requires the building the project first which can take up to 60 seconds. The API check should be something contributors can run before pushing code to verify they've captured all changes within changesets. Ideally it should take less than 30 seconds to run.

One reason for the lack of speed is the synchronous nature of the tool. Every package is checked sequentially. 

Additionally, the extractor doesn't understand how the projects are linked together and doesn't support `import('@remirror/core');` syntax.

Finally, it doesn't support `namespaces` which are used heavily within the `remirror` project for interface merging and extensibility.

- [x] Investigate `ts-morph` and how to extract API data directly from `.ts` entry points.
- [ ] Determine how long this will take to complete.
- [ ] Create relevant issues to address what needs to be done.

It depends on #389

**Comment by ifiokjr** [comment]
I've decided to temporarily use [`typedoc`](https://typedoc.org/) for documentation needs but I'll create a follow-up issue on what I'll be looking for from a documentation tool, which I haven't yet found.
